portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,43 +27,21 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-    // hard coded greeting messages
-    // private ArrayList<String> greetings;
+  // initialize user input comments to a new ArrayList
+  // <name, comments>
+  // maps all comments to each username
+  private HashMap<String, ArrayList<String>> comments;
```

Comment:
I would suggest using something like

TreeMap<timestamp, Comment>

and making a Comment class containing the name and comment text.

TreeMap will keep your comments sorted by timestamp.

This will let you start playing around before you delve into GCP data store.

Reply:
Feel free to ignore this since you're already adding data store.